pkg/webhook/policy.go

```diff
@@ -235,6 +265,27 @@ func (h *validationHandler) getDenyMessages(res []*rtypes.Result, req admission.
 					"request_username", req.AdmissionRequest.UserInfo.Username,
 				).Info("denied admission")
 			}
+			if ref != nil && ref.Name == "gatekeeper-controller-manager" {
+				annotations := map[string]string{
+					"process":            "admission",
+					"event_type":         "violation",
+					"constraint_name":    r.Constraint.GetName(),
+					"constraint_kind":    r.Constraint.GetKind(),
+					"constraint_action":  r.EnforcementAction,
+					"resource_kind":      req.AdmissionRequest.Kind.Kind,
+					"resource_namespace": req.AdmissionRequest.Namespace,
+					"resource_name":      resourceName,
+					"request_username":   req.AdmissionRequest.UserInfo.Username,
+				}
+				eventMsg := "Admission webhook \"validation.gatekeeper.sh\" denied request"
+				reason := "FailedAdmission"
+				if r.EnforcementAction == "dryrun" {
+					eventMsg = "Dryrun violation"
+					reason = "DryrunViolation"
+				}
+				h.eventRecorder.AnnotatedEventf(ref, annotations, corev1.EventTypeWarning, reason, "%s: [Namespace: %s Kind: %s Name: %s] by constraint %s %s", eventMsg, req.AdmissionRequest.Namespace, req.AdmissionRequest.Kind.Kind, resourceName, r.Constraint.GetName(), r.Msg)
```

Comment:
Do we know if `eventRecorder` has any load-reduction strategies like debouncing?